zetaclient/chains/bitcoin/bitcoin.go

```diff
@@ -224,3 +216,55 @@ func (b *Bitcoin) outboundLogger(id string) *zerolog.Logger {
 
 	return &l
 }
+
+// inboundSkipper returns a skipper function for inbound observation.
+func inboundSkipper(b *Bitcoin, app *zctx.AppContext) bool {
+	isSupported := b.observer.ChainParams().IsSupported
+	isInboundEnabled := app.IsInboundObservationEnabled()
+	isMempoolCongested := app.IsMempoolCongested()
+
+	if !isSupported || !isInboundEnabled || isMempoolCongested {
+		b.observer.Logger().
+			Chain.Debug().
+			Bool("is_supported", isSupported).
+			Bool("is_enabled", isInboundEnabled).
+			Bool("is_congested", isMempoolCongested).
+			Msg("skip inbound observation")
+		return true
+	}
+	return false
+}
+
+// outboundSkipper returns a skipper function for outbound observation.
+func outboundSkipper(b *Bitcoin, app *zctx.AppContext) bool {
```

Comment:
The name seems a bit misleading; this skipper is used for fetch_utxos and mempool as well